.github/workflows/ci.yml

```diff
@@ -145,21 +145,24 @@ jobs:
             python waf --tests --target-arch=64bit-arm all
             ls ../PyInstaller/bootloader/Windows-64bit-arm
 
-      - name: Install PyInstaller
-        run: |
-          # Update pip.
-          python -m pip install --upgrade pip setuptools wheel
+      - name: Update pip
+        run: python -m pip install --upgrade pip setuptools wheel build
 
-          # Compile bootloader
-          cd bootloader
-          python waf --tests all
-          cd ..
+      - name: Compile bootloader
+        run: cd bootloader && python waf --tests all
 
-          # Install PyInstaller.
-          pip install --progress-bar=off .[completion]
+      - name: Download dependencies
+        run: pip download --dest=dist .[completion] && rm -f dist/pyinstaller-*.whl
+        shell: bash
 
-          # Make sure the help options print.
-          python -m PyInstaller -h
+      - name: Build wheels
+        run: sh release/build-wheels
```

Comment:
Hmm, in hindsight (and after seeing this step run) - shouldn't we be building only the wheel for the runner's platform here?

This will loop through all combinations and [try to compile all non-existent bootloaders](https://github.com/pyinstaller/pyinstaller/actions/runs/11941521651/job/33286549061?pr=8891#step:17:19).

Reply:
I didn't really fancy adding another mapping of `${{ matrix.xxx }}` variations to the target names that `build-wheels` uses. It's only ~20 seconds and some messy logs we never look at.

Reply:
Fair enough. I only noticed it because I was debugging macOS pipeline, and there it was a bit more noticeable (~2 min). So it made me think we could write the step in python - this would probably allow us to programmatically infer both the bootloader directory (e.g., by importing `_shared_with_waf`) and the wheel tag, set them to `os.environ` and then run `build.main()`.

Reply:
I don't mind either way as long as this step is still effectively verifying the same wheel tags and bootloader paths as go into a release.